### ISSUE TITLE
Level-up / Advancement / Character Gen

### ISSUE THREAD

**Issue by zithith** [the-metalworks/cosmere-rpg#70]
**Is your feature request related to a problem? Please describe.**  
As the foundry implementation will be independent of Demiplane we want to provide the players a way to manage characters as well as play them.

**Describe the solution you'd like**  
Some form of process (probably a dialog) that supports users through selecting choices they are allowed at any point in their character progression.
GM grants/homebrew overrides to the core rules would be nice where possible, but probably a feature for second or third pass.

**Describe alternatives you've considered**  
Potentially we could look at doing something like a Demiplane importer, but that may not be possible and when release arrives it will ikely require the user to have made a demiplane/roll20 purchase too which seems counter-productive. An importer would still be a good feature to add on top of a built-in progression system later down the line but I feel it won't satisfy enough to replace it.

**Version**  
0.1.0

**Additional context**  
_None_

**Checklist:**
- [x] I have searched the existing issues to ensure this feature has not already been requested.

**Comment by stanavdb** [comment]
@zithith We should probably split this up into a couple of issues as well. Currently this encompasses:

- [x] Basic advancement (ability to set level & have health etc. scale)
- [x] Talent trees to be able to pick talents with prerequisites
- [ ] Character builder

**Comment by zithith** [comment]
> @zithith We should probably split this up into a couple of issues as well. Currently this encompasses...

Yeah, that absolutely crossed my mind last night, just wanted to get your take on how to split it

**Comment by zithith** [comment]
created a separate issue for the character builder to go in 1.0: #309
